src/client/src/tscParseLineProtocol.c

```diff
@@ -2362,24 +2350,30 @@ static int32_t parseSmlValue(TAOS_SML_KV *pKV, const char **idx,
       }
     }
 
+    if (start < cur) {
+      if (bufSize <= len + (cur - start)) {
```

Comment:
Expand the buffer by a factor of 2 with `realloc`, when the size of buffer is not sufficient.